### ISSUE TITLE
XR.GetAPI() throws exception in C# WinForms app

### ISSUE THREAD

**Issue by mbucchia** [dotnet/Silk.NET#726]
# Summary

Hello, this is my first time trying your package, I am using VS 2019, and version 2.11.0 of your OpenXR bindings NuGet package.

I am interested in doing is enumeration of the OpenXR capabilities for my application (no rendering etc involved, I will not even be creating a session).

When I invoke `XR.GetAPI()` from my C# application, I immediately get the following error:

```
Exception thrown: 'System.NullReferenceException' in Silk.NET.Core.dll
Object reference not set to an instance of an object.
```

# Steps to reproduce
- Platform: Desktop
- Framework Version: .NET 4.7.2
- API: OpenXR
- API Version: 1.0
1. Create a new C# WinForms project, .NET 4.7.2
2. Install NuGet package Silk.NET.OpenXR 2.11.0
3. Invoke `XR.GetAPI()` from the constructor of my WinForm.

# Comments
I am able to start up OpenXR applications and develop OpenXR applications in C++ without issues, so this is not an issue with my OpenXR loader and setup.

Please let me know if there is something incorrect with my scenario - or if you'd like me to try something to debug.

Thanks!

**Comment by Perksey** [comment]
Can you please try and get a full stack trace? (Including all the “at” bits) This isn’t a lot to go off!

**Comment by mbucchia** [comment]
Debug output:

```
'ConfigUI.exe' (CLR v4.0.30319: DefaultDomain): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_64\mscorlib\v4.0_4.0.0.0__b77a5c561934e089\mscorlib.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: DefaultDomain): Loaded 'C:\Users\mbucc\Desktop\XR\API-Layers\XR_APILAYER_NOVENDOR_nis_scaler\ConfigUI\bin\Debug\ConfigUI.exe'. Symbols loaded.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\Users\mbucc\Desktop\XR\API-Layers\XR_APILAYER_NOVENDOR_nis_scaler\ConfigUI\bin\Debug\Silk.NET.OpenXR.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\Users\mbucc\Desktop\XR\API-Layers\XR_APILAYER_NOVENDOR_nis_scaler\ConfigUI\bin\Debug\Silk.NET.Core.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\netstandard\v4.0_2.0.0.0__cc7b13ffcd2ddd51\netstandard.dll'. Module was built without symbols.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\Users\mbucc\Desktop\XR\API-Layers\XR_APILAYER_NOVENDOR_nis_scaler\ConfigUI\bin\Debug\System.Memory.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.InteropServices.RuntimeInformation\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Runtime.InteropServices.RuntimeInformation.dll'. Module was built without symbols.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\Users\mbucc\Desktop\XR\API-Layers\XR_APILAYER_NOVENDOR_nis_scaler\ConfigUI\bin\Debug\Microsoft.DotNet.PlatformAbstractions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ConfigUI.exe' (CLR v4.0.30319: ConfigUI.exe): Loaded 'C:\Users\mbucc\Desktop\XR\API-Layers\XR_APILAYER_NOVENDOR_nis_scaler\ConfigUI\bin\Debug\Microsoft.Extensions.DependencyModel.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Exception thrown: 'System.NullReferenceException' in Silk.NET.Core.dll
Object reference not set to an instance of an object.
```

Trace:

```
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.DefaultPathResolver.GetAllRuntimeIds(string currentRid, Microsoft.Extensions.DependencyModel.DependencyContext ctx)	Unknown
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.DefaultPathResolver.TryLocateNativeAssetInRuntimesFolder(string name, string baseFolder, out string result)	Unknown
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.DefaultPathResolver.CoreEnumeratePossibleLibraryLoadTargets(string name, bool noLinuxTraverse)	Unknown
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.LibraryLoader.LoadWithResolver(string name, Silk.NET.Core.Loader.PathResolver pathResolver)	Unknown
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.LibraryLoader.TryLoadNativeLibrary(string[] names, Silk.NET.Core.Loader.PathResolver pathResolver, out nint result)	Unknown
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.LibraryLoader.TryLoadNativeLibrary(string name, Silk.NET.Core.Loader.PathResolver pathResolver, out nint result)	Unknown
 	Silk.NET.Core.dll!Silk.NET.Core.Loader.LibraryLoader.LoadNativeLibrary(string name, Silk.NET.Core.Loader.PathResolver pathResolver)	Unknown
 	Silk.NET.OpenXR.dll!Silk.NET.OpenXR.XR.GetApi()	Unknown
>	ConfigUI.exe!ConfigUI.Form1.InitXr() Line 39	C#
 	ConfigUI.exe!ConfigUI.Form1.Form1() Line 23	C#
 	ConfigUI.exe!ConfigUI.Program.Main() Line 19	C#
```

I'm onto something now - I had assumed that the NuGet package would include the OpenXR loader, but it looks like it doesn't. Will try to copy the DLL manually now and see if it works.

**Comment by mbucchia** [comment]
Ahhh that was it!

Surely the error message can be improved. Surpisingly to me, I tried a console app project instead of WinForms and I got a much better error which actually told me what the problem was:

```
Exception thrown: 'System.IO.FileNotFoundException' in Silk.NET.Core.dll
An unhandled exception of type 'System.IO.FileNotFoundException' occurred in Silk.NET.Core.dll
Could not find or load the native library: openxr_loader.dll Attempted: "openxr_loader.dll", "C:\Users\mbucchianeri\source\repos\ConsoleApp1\bin\Debug\net5.0\openxr_loader.dll", "C:\Users\mbucchianeri\source\repos\ConsoleApp1\bin\Debug\net5.0\openxr_loader.dll"
```

**Comment by mbucchia** [comment]
Thanks for the prompt response btw - I'll let you close as-is if you want - however still suggesting to improve the error message (or at least identify why I did not get the more clear error message with a WinForms app).

**Comment by Perksey** [comment]
Yeah the null exception is not intended, will work on fixing this soon.

**Comment by Perksey** [comment]
I was about to post a comment explaining that I can't reproduce this, but then I realised we already had a community PR to fix the NullReferenceException bug! (#721)

Closing this now, thanks for the report :)
